### PR TITLE
Remove BasicUI class

### PR DESCRIPTION
**Pull request by simonhong**
No functional change. Just cleanup.

Resolves https://github.com/brave/brave-browser/issues/13955

## Submitter Checklist:

- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed).
- [ ] Requested a security/privacy review as needed.

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: browser/ui/webui/brave_web_ui_controller_factory.cc [hunk 1]
```diff
@@ -60,14 +60,7 @@ namespace {
 typedef WebUIController* (*WebUIFactoryFunction)(WebUI* web_ui,
                                                  const GURL& url);
 
-// Template for defining WebUIFactoryFunction.
-template<class T>
```
Comment (simonhong) [inline browser/ui/webui/brave_web_ui_controller_factory.cc:64]: We don't need template here. So far we only have one specialized template.

File: browser/ui/webui/webcompat_reporter_ui.cc [hunk 2]
```diff
@@ -36,11 +36,12 @@ class WebcompatReporterDOMHandler : public content::WebUIMessageHandler {
   DISALLOW_COPY_AND_ASSIGN(WebcompatReporterDOMHandler);
 };
 
-WebcompatReporterDOMHandler::WebcompatReporterDOMHandler()
+WebcompatReporterDOMHandler::WebcompatReporterDOMHandler(
+    scoped_refptr<network::SharedURLLoaderFactory> loader_factory)
     : uploader_(std::make_unique<brave::WebcompatReportUploader>(
-          g_browser_process->shared_url_loader_factory())) {}
```
Comment (simonhong) [inline browser/ui/webui/webcompat_reporter_ui.cc:41]: we should avoid using global loader.

File: browser/ui/webui/webui_utils.h [hunk 3]
```diff
@@ -0,0 +1,26 @@
+/* Copyright 2019 The Brave Authors. All rights reserved.
```
Comment (iefremov) [inline browser/ui/webui/webui_utils.h:1]: 2021

File: browser/ui/webui/webui_utils.h [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+#include <string>
+
+struct GritResourceMap;
+
+namespace content {
+class WebUI;
+class WebUIDataSource;
+}  // namespace content
+
+content::WebUIDataSource* CreateAndAddWebUIDataSource(
```
Comment (iefremov) [inline browser/ui/webui/webui_utils.h:18]: I would add a comment for this one

**Review by iefremov — APPROVED** [review]
Thanks @simonhong !

**Review by bsclifton — APPROVED** [review]
++

**Review by zenparsing — APPROVED** [review]
Rewards WebUI changes look good. ❤️

**Pull request merged**